### DISCUSSION TITLE
Adding new default fields

### DISCUSSION THREAD

**Discussion by LTrochain** [PrestaSafe/prettyblocks#119] [Q&A] [answered]
Hi, 
Small prestashop developper here, 

I would like to know if there is some best practices to add new default fields for blocks.
As an example, I would like to add mobile and desktop visibility everyblocks :
~          'container' => true,
~          'load_ajax' => false,
~          'bg_color' => '',
+          'display_desktop' => true,
+          'display_mobile' => true,

Any tips ? Thanks :)

**Comment by PrestaSafe** [comment, marked as answer]
Hi, you can add it here https://github.com/PrestaSafe/prettyblocks/blob/d9f60570c194cf768863958710a5419dcfc2e3da/classes/PrettyBlocksModel.php#L567 
And make a pull request if you want. 
It could be hookable, i'll try to make that soon :)

**Reply by LTrochain** [reply]
Thanks for your support !
I added the custom fields inside PrettyBlocksModel.php 
Tweaked a little RightPanel.vue
Wrote some translations into AdminThemeManagerController.php 
Rebuilt everything.. And tadaa, here are the new fields 🎉 

![image](https://github.com/PrestaSafe/prettyblocks/assets/94692231/de0e2b4a-689d-4588-a32e-8793a0894a01)

**Reply by PrestaSafe** [reply]
Nice ! But if there is an update, your fields will be removed. 
i've opened an issue for that: https://github.com/PrestaSafe/prettyblocks/issues/123 

We will add this soon
